src/Symfony/Component/HttpKernel/Event/FilterControllerArgumentsEvent.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Event;
+
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Allows filtering of controller arguments.
+ *
+ * You can call getController() to retrieve the controller and getArguments
+ * to retrieve the current arguments. With setArguments() you can replace
+ * arguments that are used to call the controller.
+ *
+ * Arguments set in the event must be compatible with the signature of the
+ * controller.
+ *
+ * @author Christophe Coevoet <stof@notk.org>
+ */
+class FilterControllerArgumentsEvent extends KernelEvent
+{
+    private $controller;
+    private $arguments;
+
+    public function __construct(HttpKernelInterface $kernel, callable $controller, array $arguments, Request $request, $requestType)
+    {
+        parent::__construct($kernel, $request, $requestType);
+
+        $this->controller = $controller;
+        $this->arguments = $arguments;
+    }
+
+    /**
+     * Returns the current controller.
+     *
+     * @return callable
+     */
+    public function getController()
+    {
+        return $this->controller;
+    }
+
+    /**
+     * @return array
+     */
+    public function getArguments()
+    {
+        return $this->arguments;
+    }
+
+    /**
+     * @param array $arguments
+     */
+    public function setArguments(array $arguments)
+    {
+        $this->arguments = $arguments;
+    }
```

Comment:
Imagine a case where I get 3 arguments:
```php
[ 0=> 'foo', 1 => 'bar', 2 => 'baz' ]
```
I want to update the last element but I did `$arguments[3] = 'foo';` instead of `$arguments[2] = 'foo';`, I would end up with 1 argument more and no clear trace of where what happened.

Another issue is that you can't really do much at this point with those arguments. You don't have access to the metadata so if you want to do _anything_ with introspection, you would have to (again) create the metadata or do it manually with reflection. Until there's a cached variant, the first is really not something I would advice.

This would not be the correct hook to modify the arguments. I would rather get rid of the `setController()` in the `FilterControllerEvent` as well and use a `ControllerFactory` for the controller event (on my todo list). This means you have 1 fixed point of turning `_controller` (or whatever else) into something callable, just like you have 1 fixed point of setting the argument at this point.